app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt

```diff
@@ -213,34 +214,66 @@ class TrackingProtectionFragment : PreferenceFragmentCompat() {
 
         customCookies.onPreferenceChangeListener = object : SharedPreferenceUpdater() {
             override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
-                    updateTrackingProtectionPolicy()
                 customCookiesSelect.isVisible = !customCookies.isChecked
-                return super.onPreferenceChange(preference, newValue)
+                return super.onPreferenceChange(preference, newValue).also {
+                    updateTrackingProtectionPolicy()
+                }
             }
         }
 
         customTracking.onPreferenceChangeListener = object : SharedPreferenceUpdater() {
             override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
-                    updateTrackingProtectionPolicy()
                 customTrackingSelect.isVisible = !customTracking.isChecked
-                return super.onPreferenceChange(preference, newValue)
+                return super.onPreferenceChange(preference, newValue).also {
+                    updateTrackingProtectionPolicy()
+                }
             }
         }
 
         customCookiesSelect.onPreferenceChangeListener = object : StringSharedPreferenceUpdater() {
             override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
-                updateTrackingProtectionPolicy()
-                val newValueEntry = (preference as DropDownListPreference).findEntry(key = newValue)
-                return super.onPreferenceChange(preference, newValueEntry)
```

Comment:
This whole ETP thing is really complex and hard to follow.  I think we were mistaken before @mcarare, this is working fine without any of the preference key/value mapping.

Reply:
Yes, I think it handles that internally. The initial problem was using `SharedPreferenceUpdater` that only handles booleans, not string. With the new `StringSharedPreferenceUpdater` it works without the mapping.